tests/arch/arm/arm_runtime_nmi/src/arm_runtime_nmi.c

```diff
@@ -60,6 +60,10 @@ void test_arm_runtime_nmi(void)
 
 	/* Trigger NMI: Should fire immediately */
 	SCB->ICSR |= SCB_ICSR_NMIPENDSET_Msk;
+
+	k_sleep(K_MSEC(1000));
```

Comment:
I think the delay can be shorter here. Or, try to have no delay here. Because the NMI should react right away.

Reply:
tried 10 ms, the result is the same

Reply:
Delay does not matter, maybe:
```
	__ISB();
	__DSB();
```
But other than that you can just run the assert immediately.

Reply:
OK, will remove the delay.